src/runtime.cpp

```diff
@@ -109,6 +110,10 @@ namespace hpx
 #include <stdlib.h>
 #include <string.h>
 
+#if (HPX_HAVE_STACKOVERFLOW_DETECTION == "ON")
```

Comment:
This should be either:

    #ifdef HPX_HAVE_STACKOVERFLOW_DETECTION 

or

    #if defined(HPX_HAVE_STACKOVERFLOW_DETECTION )

I'm surprised this compiled for you in the first place. Circle-ci at least is not happy (see: https://circleci.com/gh/STEllAR-GROUP/hpx/7525).